src/julia_threads.h

```diff
@@ -74,6 +77,17 @@ typedef struct _jl_tls_states_t {
 #  define JL_CPU_WAKE_NOOP 1
 #endif
 
+// Copied from libuv. Add `JL_CONST_FUNC` so that the compiler
+// can optimize this better.
+static inline unsigned long JL_CONST_FUNC jl_thread_self(void)
+{
+#ifdef _OS_WINDOWS_
+    return (unsigned long)GetCurrentThreadId();
+#else
+    return (unsigned long)pthread_self();
+#endif
+}
```

Comment:
Is breaking the ``libuv`` abstraction a good idea? I have a PR open for upstream ``libuv`` for a patch I made to our fork to handle thread affinity in the library so that we don't use pthreads directly.